Sources/MapboxNavigation/RequestDownload.swift

```diff
@@ -6,14 +6,14 @@ enum DownloadError: Error {
     case noImageData
 }
 
-protocol ImageDownload: URLSessionDataDelegate {
+protocol RequestDownload: URLSessionDataDelegate {
     init(request: URLRequest, in session: URLSession)
-    func addCompletion(_ completion: @escaping ImageDownloadCompletionBlock)
+    func addCompletion(_ completion: @escaping CachedResponseCompletionBlock)
```

Comment:
Update the `ImageDownload` to a more generic `RequestDownload`. The repository will be responsible for generating urlRequest and converting data to image.